src/analyze/analyze-chid.c

```diff
@@ -164,11 +152,34 @@ int verb_chid(int argc, char *argv[], void *userdata) {
 
                 size--;
 
-                smbios_fields[f] = utf8_to_utf16(buf, size);
-                if (!smbios_fields[f])
+                (*fields)[f] = utf8_to_utf16(buf, size);
+                if (!(*fields)[f])
                         return log_oom();
         }
 
+        return 0;
+}
+
+int verb_chid(int argc, char *argv[], void *userdata) {
+
+        _cleanup_(table_unrefp) Table *table = NULL;
+        int r;
+
+        if (detect_container() > 0)
+                return log_error_errno(SYNTHETIC_ERRNO(ENOTRECOVERABLE), "Container environments do not have SMBIOS.");
+
+        table = table_new("type", "input", "chid");
+        if (!table)
+                return log_oom();
+
+        (void) table_set_align_percent(table, table_get_cell(table, 0, 0), 100);
+        (void) table_set_align_percent(table, table_get_cell(table, 0, 1), 50);
+
+        _cleanup_(smbios_fields_free) char16_t* smbios_fields[_CHID_SMBIOS_FIELDS_MAX] = {};
+        r = smbios_fields_acquire(&smbios_fields);
```

Comment:
Hm? Why not to pass `smbios_fields` as is?

Reply:
no idea really, not sure what i awas smoking ;-)